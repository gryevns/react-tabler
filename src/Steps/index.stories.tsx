import React from "react";

import { colors } from "../common";
import Steps from "../Steps";
import Tooltip from "../Tooltip";

export default {
    title: "Steps",
};

export const Default = () => (
    <Steps>
        <Steps.Step>Step 1</Steps.Step>
        <Steps.Step>Step 2</Steps.Step>
        <Steps.Step>Step 3</Steps.Step>
        <Steps.Step>Step 4</Steps.Step>
    </Steps>
);

export const Tooltips = () => (
    <Steps>
        <Tooltip placement="bottom" content="one">
            <Steps.Step>Step 1</Steps.Step>
        </Tooltip>
        <Tooltip placement="bottom" content="two">
            <Steps.Step>Step 2</Steps.Step>
        </Tooltip>
        <Tooltip placement="bottom" content="three">
            <Steps.Step>Step 3</Steps.Step>
        </Tooltip>
    </Steps>
);

export const Colors = () =>
    colors.map((color) => (
        <Steps color={color}>
            <Steps.Step>Step 1</Steps.Step>
            <Steps.Step>Step 2</Steps.Step>
            <Steps.Step>Step 3</Steps.Step>
            <Steps.Step>Step 4</Steps.Step>
        </Steps>
    ));

export const Basic = () => (
    <Steps>
        <Steps.Step />
        <Steps.Step />
        <Steps.Step active />
        <Steps.Step />
    </Steps>
);

export const Numbers = () => (
    <Steps numbers>
        <Steps.Step />
        <Steps.Step active />
        <Steps.Step />
        <Steps.Step />
    </Steps>
);
