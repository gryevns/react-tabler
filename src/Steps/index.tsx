import React from "react";
import cn from "classnames";
import { Color } from "../common";

export interface StepProps extends React.AnchorHTMLAttributes<HTMLElement> {
    active?: boolean;
    children?: React.ReactNode;
}

const Step = React.forwardRef(
    (
        { active, children, ...props }: StepProps,
        ref: React.Ref<HTMLAnchorElement>,
    ) => {
        const classes = cn("step-item", { active });
        return (
            <a ref={ref} {...props} className={classes}>
                {children}
            </a>
        );
    },
);

export interface StepsProps {
    className?: string;
    children: React.ReactNode;
    color?: Color;
    numbers?: boolean;
}

const Steps = ({ className, children, color, numbers }: StepsProps) => {
    const classes = cn(
        "steps",
        {
            [`steps-${color}`]: !!color,
            "steps-counter": numbers,
        },
        className,
    );
    return <div className={classes}>{children}</div>;
};

Steps.Step = Step;

export default Steps;
