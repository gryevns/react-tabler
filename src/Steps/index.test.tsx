import React from "react";
import { shallow } from "enzyme";

import Steps from "./index";

describe("Steps", () => {
    it("should match steps snapshot", () => {
        const wrapper = shallow(
            <Steps color="red">
                <Steps.Step>Step 1</Steps.Step>
                <Steps.Step active>Step 2</Steps.Step>
                <Steps.Step>Step 3</Steps.Step>
                <Steps.Step>Step 4</Steps.Step>
            </Steps>,
        );
        expect(wrapper).toMatchSnapshot();
    });
    it("should match step snapshot", () => {
        const wrapper = shallow(<Steps.Step>Step 1</Steps.Step>);
        expect(wrapper).toMatchSnapshot();
    });
    it("should match active step snapshot", () => {
        const wrapper = shallow(<Steps.Step active>Step 1</Steps.Step>);
        expect(wrapper).toMatchSnapshot();
    });
});
