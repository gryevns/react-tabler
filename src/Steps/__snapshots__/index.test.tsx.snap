// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps should match active step snapshot 1`] = `
<a
  className="step-item active"
>
  Step 1
</a>
`;

exports[`Steps should match step snapshot 1`] = `
<a
  className="step-item"
>
  Step 1
</a>
`;

exports[`Steps should match steps snapshot 1`] = `
<div
  className="steps steps-red"
>
  <ForwardRef>
    Step 1
  </ForwardRef>
  <ForwardRef
    active={true}
  >
    Step 2
  </ForwardRef>
  <ForwardRef>
    Step 3
  </ForwardRef>
  <ForwardRef>
    Step 4
  </ForwardRef>
</div>
`;
